devito/operator.py

```diff
@@ -188,6 +192,36 @@ def __init__(self, expressions, **kwargs):
 
     # Compilation
 
+    def _add_implicit(self, expressions):
+        """
+        Create and add to the expression tree any associated implicit
+        expressions.
+        """
+        updated_expressions = []
+        for e in expressions:
+            subdomain = e.subdomain
+            if bool(subdomain):
+                try:
+                    dat = e._subdomain._ie_dat
+                except AttributeError:
+                    dat = None
+                if bool(dat):
```

Comment:
no need for this ``if`` ?

I think this piece of code becomes more natural if you take the current if's body and move it inside the ``try``. Then, in the ``except``, you just put ``pass``

Reply:
Sure, sounds good.